frontend/compositions/visualizations/map/marker-layer.tsx

```diff
@@ -1,41 +1,140 @@
+
+    // popup.moveTo({x: x, y: y})
+    // popup.setHeader(markerDescription.label)
+    // popup.setBody("this is a marker popup")
+    // popup.show()
+  }
+
+  const handleMouseMove = (event: MouseEvent) => {
+    event.stopPropagation()
+  }
+
+  const handleMouseLeave = () => {
+    console.log("mosue leave makrer")
+    // popup.hide()
+  }
+
   return (
-    <svg viewBox="0, 0, 1200, 700" className={styles.markerLayer} height={"100%"} width={"100%"}>
+    <svg
+      viewBox="0, 0, 1200, 700"
+      id={"marker-layer"}
+      className={styles.markerLayer}
+      height={"100%"}
+      width={"100%"}>
       {markersData &&
         markersData.map((c, i) => {
-          return <CircleMarker markerDescription={c} key={`${i}${c.dataPoint}`} />
+          return (
+            <>
+              <Marker
+                type={"circle"}
+                markerDescription={c}
+                key={`${i}${c.dataPoint}`}
+                handleMouseOver={handleMouseOver}
+                handleMouseLeave={handleMouseLeave}
+                handleMouseMove={handleMouseMove}
+              />
+            </>
+          )
         })}
     </svg>
   )
 }
 
-export interface CircleMarkerProps {
+export interface MarkerProps {
   markerDescription: MarkerDescription
+  mousePosition?: Coord
+  handleMouseOver(event: any, markerDescription: MarkerDescription): void
+  handleMouseLeave(): void
+  handleMouseMove(event: any, markerDescription: MarkerDescription): void
+  type?: "circle" | "square"
+  transformScale?: number
 }
 
-export function CircleMarker(props: CircleMarkerProps) {
-  const { markerDescription: c } = props
+export function Marker(props: MarkerProps) {
+  const {
+    type,
+    markerDescription,
+    handleMouseLeave,
+    handleMouseMove,
+    handleMouseOver,
+    transformScale
+  } = props
+
+  const coordinate: Coord = markerDescription.geoCenter
+
+  return (
+    <g
+      type={type}
+      {...props}
+      onClick={() => console.log(markerDescription.label)}
+      onMouseOver={(e) => handleMouseOver(e, markerDescription)}
+      onMouseMove={(e) => handleMouseMove(e, markerDescription)}
+      onMouseOut={handleMouseLeave}>
+      {type === "circle" && <CircleMarker position={coordinate} />}
+      {type === "square" && <SquareMarker position={coordinate} />}
+    </g>
+  )
+}
+
+export function CircleMarker(props: { position: Coord; fill?: string }) {
+  const {
+    position: { x, y },
+    fill
+  } = props
+  const r = 10
+  const cx = x
+  const cy = y
+
   return (
     <circle
-      className={styles.circleMarker}
-      cx={c.geoCenter.x}
-      cy={c.geoCenter.y}
-      r={c.dataPoint}
-      onClick={() => console.log(c.label)}
+      className={"marker " + styles.circleMarker}
+      cx={cx}
+      cy={cy}
+      r={r}
+      fill={fill || "var(--darkBlue)"}
+    />
+  )
+}
+
+export function SquareMarker(props: { position: Coord; fill?: string }) {
+  const {
+    position: { x, y },
+    fill
+  } = props
```

Comment:
This can be destructured directly in the function parenthesis - and the type can be factored out into an interface for brevity/give us an opportunity to name it informatively!

Reply:
Same seemingly with the other places we're taking props in this file!